gremlin-driver/src/test/java/org/apache/tinkerpop/gremlin/driver/handler/HttpGremlinResponseStreamDecoderTest.java

```diff
@@ -0,0 +1,102 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.tinkerpop.gremlin.driver.handler;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.ByteBufAllocator;
+import io.netty.channel.embedded.EmbeddedChannel;
+import io.netty.handler.codec.TooLongFrameException;
+import io.netty.handler.codec.http.DefaultFullHttpResponse;
+import io.netty.handler.codec.http.DefaultHttpHeaders;
+import io.netty.handler.codec.http.FullHttpResponse;
+import io.netty.handler.codec.http.HttpObjectAggregator;
+import io.netty.handler.codec.http.HttpResponseStatus;
+import io.netty.handler.codec.http.HttpServerCodec;
+import io.netty.handler.codec.http.HttpVersion;
+import java.util.Collections;
+import org.apache.commons.lang3.RandomStringUtils;
+import org.apache.tinkerpop.gremlin.util.message.ResponseMessage;
+import org.apache.tinkerpop.gremlin.util.ser.SerializationException;
+import org.apache.tinkerpop.gremlin.util.ser.Serializers;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+public class HttpGremlinResponseStreamDecoderTest {
+
+    @Test
+    public void shouldSucceedIfResponseSizeUnderMaxResponseContentLength() throws SerializationException {
+        final String content = "this response is smaller than the max allowed";
+        final FullHttpResponse httpResponse = createResponse(content);
+        final EmbeddedChannel testChannel = initializeChannel(httpResponse.content().readableBytes() + 1);
+
+        testChannel.writeInbound(httpResponse);
+        final ResponseMessage inbound = testChannel.readInbound();
+        assertEquals(content, inbound.getResult().getData().get(0));
+    }
+
+    @Test
+    public void shouldSucceedIfResponseSizeEqualToMaxResponseContentLength() throws SerializationException {
+        final String content = "this response is equal to the max allowed";
+        final FullHttpResponse httpResponse = createResponse(content);
+        final EmbeddedChannel testChannel = initializeChannel(httpResponse.content().readableBytes());
+
+        testChannel.writeInbound(httpResponse);
+        final ResponseMessage inbound = testChannel.readInbound();
+        assertEquals(content, inbound.getResult().getData().get(0));
+    }
+
+    @Test
+    public void shouldSucceedIfMaxResponseContentLengthZero() throws SerializationException {
+        final String largeResponse = RandomStringUtils.random(3000);
+        final FullHttpResponse httpResponse = createResponse(largeResponse);
+        final EmbeddedChannel testChannel = initializeChannel(0);
+
+        testChannel.writeInbound(httpResponse);
+        final ResponseMessage inbound = testChannel.readInbound();
+        assertEquals(largeResponse, inbound.getResult().getData().get(0));
+    }
+
+    @Test
+    public void shouldSucceedIfResponseSizeLargerThanMaxResponseContentLength() throws SerializationException {
```

Comment:
The name of this test is bit misleading. `shouldSucceed` when the assert is that an exception occurs.

Reply:
Changed to 'shouldThrow'